### PR TITLE
[dev-launcher] move session fns from dev-menu to dev-launcher i

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

The new dev-launcher is now responsible for the user auth and session management - but these functions were still living in the dev-menu package - this PR aims to move the functionality into `dev-launcher` in an effort to clean things up

Android implementation is not yet done

# How

Moved all the functions related to user auth from dev menu into the dev-launcher package - in order to do so I had to reimplement a few things in objective-c 

# Test Plan

- unit tests should still pass 
- login / logout should work as expected in the app

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-dev-launcher/bundle/native-modules/DevLauncherInternal.ts [hunk 1]
```diff
@@ -44,3 +51,117 @@ export async function copyToClipboardAsync(content: string): Promise<null> {
 export const clientUrlScheme = DevLauncher.clientUrlScheme;
 export const installationID = DevLauncher.installationID;
 export const isDevice = !!DevLauncher.isDevice;
+
```
Comment (esamelson) [inline packages/expo-dev-launcher/bundle/native-modules/DevLauncherInternal.ts:54]: do you think there is value in keeping the auth session / web browser stuff as a separate native module, even if it lives inside the dev launcher package?

File: packages/expo-dev-menu/ios/DevMenuExpoSessionDelegate.swift [hunk 2]
```diff
@@ -1,55 +0,0 @@
...
-
-  private func setSesssionSecret(_ sessionSecret: String?) {
-    let wasLoggedIn = manager.expoApiClient.isLoggedIn()
-    manager.expoApiClient.setSessionSecret(sessionSecret)
-    let isLoggedIn = manager.expoApiClient.isLoggedIn()
-
-    if !wasLoggedIn && isLoggedIn {
-      manager.sendEventToDelegateBridge(DevMenuExpoSessionDelegate.userLoginEvent, data: nil)
-    } else if wasLoggedIn && !isLoggedIn {
-      manager.sendEventToDelegateBridge(DevMenuExpoSessionDelegate.userLogoutEvent, data: nil)
```
Comment (esamelson) [inline packages/expo-dev-menu/ios/DevMenuExpoSessionDelegate.swift:52]: are these events unused now?

**Review by esamelson — COMMENTED** [review]
generally looks really good, just a few thoughts/questions

File: packages/expo-dev-launcher/bundle/apiClient.ts [hunk 3]
```diff
@@ -1,8 +1,6 @@
 import { GraphQLClient } from 'graphql-request';
 
-export const apiEndpoint = __DEV__
-  ? `https://staging.exp.host/--/graphql`
-  : `https://exp.host/--/graphql`;
-export const websiteOrigin = __DEV__ ? 'https://staging.expo.dev' : 'https://expo.dev';
+export const apiEndpoint = __DEV__ ? `https://exp.host/--/graphql` : `https://exp.host/--/graphql`;
+export const websiteOrigin = __DEV__ ? 'https://expo.dev' : 'https://expo.dev';
```
[the remark this answers is not in the document]
Comment (ajsmth) [reply packages/expo-dev-launcher/bundle/apiClient.ts:4]: will do! this was intentional as updating the api calls ended up being a bit more complicated - better suited for a different PR

File: packages/expo-dev-launcher/bundle/native-modules/DevLauncherInternal.ts (hunk 1 above)
Comment (ajsmth) [reply packages/expo-dev-launcher/bundle/native-modules/DevLauncherInternal.ts:54]: I think that makes a lot of sense

File: packages/expo-dev-menu/ios/DevMenuExpoSessionDelegate.swift (hunk 2 above)
Comment (ajsmth) [reply packages/expo-dev-menu/ios/DevMenuExpoSessionDelegate.swift:52]: yes that's correct

File: packages/expo-dev-launcher/bundle/native-modules/DevMenu.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ export async function queryMyProjectsAsync(): Promise<any> {
 
 export async function queryDevSessionsAsync(installationID?: string): Promise<DevSession[]> {
   const data = await DevMenu.queryDevSessionsAsync(installationID);
-
+
```
Comment (lukmccall) [inline packages/expo-dev-launcher/bundle/native-modules/DevMenu.ts:17]:
```suggestion

```

File: packages/expo-dev-launcher/ios/EXDevLauncherAuth.h [hunk 5]
```diff
@@ -0,0 +1,17 @@
+//
+//  EXDevLauncherAuth.h
+//  Pods
+//
+//  Created by andrew on 2022-02-02.
+//
```
Comment (lukmccall) [inline packages/expo-dev-launcher/ios/EXDevLauncherAuth.h:6]:
```suggestion
// Copyright 2015-present 650 Industries. All rights reserved.
```

File: packages/expo-dev-launcher/ios/EXDevLauncherAuth.m [hunk 6]
```diff
@@ -0,0 +1,149 @@
+//
+//  EXDevLauncherAuth.m
+//  Pods
+//
+//  Created by andrew on 2022-02-02.
+//
```
Comment (lukmccall) [inline packages/expo-dev-launcher/ios/EXDevLauncherAuth.m:6]:
```suggestion
// Copyright 2015-present 650 Industries. All rights reserved.
```

File: packages/expo-dev-launcher/ios/EXDevLauncherAuth.m [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+#import <SafariServices/SafariServices.h>
+
+#if __has_include(<EXDevLauncher/EXDevLauncher-Swift.h>)
+// For cocoapods framework, the generated swift header will be inside EXDevLauncher module
+#import <EXDevLauncher/EXDevLauncher-Swift.h>
+#else
+#import <EXDevLauncher-Swift.h>
+#endif
+
+
```
Comment (lukmccall) [inline packages/expo-dev-launcher/ios/EXDevLauncherAuth.m:20]:
```suggestion
```

File: packages/expo-dev-launcher/ios/EXDevLauncherAuth.m [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+@import EXDevMenu;
+
+
```
Comment (lukmccall) [inline packages/expo-dev-launcher/ios/EXDevLauncherAuth.m:23]:
```suggestion
```

File: packages/expo-dev-launcher/ios/EXDevLauncherAuth.m [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+NSString *DEV_LAUNCHER_DEFAULT_SCHEME = @"expo-dev-launcher";
+
+
```
Comment (lukmccall) [inline packages/expo-dev-launcher/ios/EXDevLauncherAuth.m:26]:
```suggestion
```

File: packages/expo-dev-launcher/ios/EXDevLauncherAuth.m [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+@end
+
+@implementation EXDevLauncherAuth
+
++ (NSString *)moduleName
+{
+  return @"EXDevLauncherAuth";
+}
+
+
```
Comment (lukmccall) [inline packages/expo-dev-launcher/ios/EXDevLauncherAuth.m:42]:
```suggestion
```

File: packages/expo-dev-launcher/ios/EXDevLauncherAuth.m [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+  }
+}
+
+- (void)flowDidFinish
+{
+  self.redirectResolve = nil;
+  self.redirectReject = nil;
+}
+
+
```
Comment (lukmccall) [inline packages/expo-dev-launcher/ios/EXDevLauncherAuth.m:92]:
```suggestion
```

File: packages/expo-dev-launcher/ios/EXDevLauncherAuth.m [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+/**
+ * Helper that is used in openBrowserAsync and openAuthSessionAsync
+ */
+- (BOOL)initializeWebBrowserWithResolver:(RCTPromiseResolveBlock)resolve andRejecter:(RCTPromiseRejectBlock)reject {
+
```
Comment (lukmccall) [inline packages/expo-dev-launcher/ios/EXDevLauncherAuth.m:97]:
```suggestion
```

**Review by lukmccall — APPROVED** [review]

**Review by esamelson — APPROVED** [review]
🎉

**Pull request merged**